### PR TITLE
add post model

### PR DESCRIPTION
**Pull request by abdelrahmanabdelnabi**
I created the initial implementation of the post model. No support for images in posts yet, I am thinking of using [paperclip](https://github.com/thoughtbot/paperclip) to add support for images.

### CODE REVIEW

File: api/db/migrate/20171217200945_adding_foreign_key_contraints.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddingForeignKeyContraints < ActiveRecord::Migration[5.1]
```
Comment (Youssefares) [inline api/db/migrate/20171217200945_adding_foreign_key_contraints.rb]: We can have a better name for this migration.

File: api/app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Post < ApplicationRecord
+  belongs_to :user
```
Comment (Youssefares) [inline api/app/models/post.rb:4]: I believe  you need to add `has_many: :posts` to the User model as well.

**Review by Youssefares — CHANGES_REQUESTED** [review]
Paperclip is solid. Another solution is storing base64 string and using https://github.com/carrierwaveuploader/carrierwave I think.

Never mingled with these myself so research and make a decision :wink: 
Minor changes requested :+1: looks good to me.

**Comment by abdelrahmanabdelnabi** [comment]
requested changes completed and resolved conflicts with master

**Pull request merged**